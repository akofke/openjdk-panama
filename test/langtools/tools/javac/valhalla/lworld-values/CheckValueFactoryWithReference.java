/*
 * @test /nodynamiccopyright/
 * @summary Do not allow mismatched instantiation syntax between value & reference types.
 *
 * @compile/fail/ref=CheckValueFactoryWithReference.out -XDrawDiagnostics -XDdisallowValueConstructors CheckValueFactoryWithReference.java
 */

final class CheckValueFactoryWithReference {
    final Object o = Object.default;
    value final class Point { int x = 10; }
    Point p = new Point();
}
