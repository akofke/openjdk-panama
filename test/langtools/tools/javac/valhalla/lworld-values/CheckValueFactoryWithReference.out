CheckValueFactoryWithReference.java:9:28: compiler.err.make.default.with.nonvalue
CheckValueFactoryWithReference.java:11:15: compiler.err.garbled.value.reference.instantiation
2 errors
